pkg/csi/service/vanilla/controller.go

```diff
@@ -351,6 +351,37 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 	}
 	volSizeMB := int64(common.RoundUpSize(volSizeBytes, common.MbInBytes))
 
+	// Check if the feature state of block-volume-snapshot is enabled
+	isBlockVolumeSnapshotEnabled := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.BlockVolumeSnapshot)
+	// Check if requested volume size and source snapshot size matches
+	volumeSource := req.GetVolumeContentSource()
+	var contentSourceSnapshotID string
+	if isBlockVolumeSnapshotEnabled && volumeSource != nil {
+		sourceSnapshot := volumeSource.GetSnapshot()
+		if sourceSnapshot == nil {
+			return nil, logger.LogNewErrorCode(log, codes.InvalidArgument, "unsupported VolumeContentSource type")
```

Comment:
This looks likes something is unset, but the error message states it's an unsupported type.

Reply:
Not really. As long as `volumeSource != nil`, it can only be one of the two types, `volume` and `snapshot`. 
```
sourceSnapshot := volumeSource.GetSnapshot()
```
The line above is used to cast the type into snapshot. I don't see any ambiguity here.

Reply:
Got it. Thanks.